File: 20241022/jiyeon/동영상재생기.py
def solution(video_len, pos, op_start, op_end, commands):
    answer = ''
    video_len_s = toSecond(video_len)
    pos_s = toSecond(pos)
    op_start_s = toSecond(op_start)
    op_end_s = toSecond(op_end)

    # 오프닝 건너뛰기
    if (op_start_s <= pos_s <= op_end_s):
        pos_s = op_end_s

    for command in commands:
        if (command == "prev"):
            pos_s -= 10
            if (pos_s < 0):
                pos_s = 0
        elif (command == "next"):
            pos_s += 10
            if (pos_s > video_len_s):
                pos_s = video_len_s

        # 오프닝 건너뛰기
        if (op_start_s <= pos_s <= op_end_s):
            pos_s = op_end_s
    answer = toTimeString(pos_s)
    return answer


def toSecond(time):
    m, s = map(int, time.split(":"))
    return m * 60 + s


def toTimeString(second):
    m = second // 60
    s = second % 60
    filled_m = "{:02}".format(m)
    filled_s = "{:02}".format(s)
    return filled_m + ":" + filled_s

File: 20241022/donggyu/동영상_재생기.java
class Solution {
    public String solution(String video_len, String pos, String op_start, String op_end, String[] commands) {
        
		int nowPos = toSecond(pos);
		
		for(String o : commands) {
			nowPos = checkJump(op_start, op_end, nowPos);
			if("prev".equals(o)) {
				nowPos = prev(nowPos);
			} else if("next".equals(o)) {
				nowPos = next(nowPos, toSecond(video_len));
			}		
		}
        nowPos = checkJump(op_start, op_end, nowPos);
		return toMinute(nowPos);
    }
    
   static int checkJump(String op_start, String op_end, int nowPos) {
		int start = toSecond(op_start);
		int end = toSecond(op_end);
		if(start<=nowPos && nowPos <= end){
			nowPos = end;
		}
       
		return nowPos;
	}
	static int next(int time, int endTime) {
		int returnTime = time+10;
		if(returnTime>endTime)
			returnTime = endTime;
		
		return returnTime;
	}
	
	static int prev(int time) {
		int returnTime = time-10;
		if(returnTime<0)
			returnTime = 0;

		return returnTime;
	}
    
	static int toSecond(String time) {
		int second = 0;
		String[] timeList = time.split(":");
		second = Integer.parseInt(timeList[0])*60+Integer.parseInt(timeList[1]);
		return second;
	}

	static String toMinute(int time) {
		int minute = time/60;
		int second = time%60;
		return String.format("%02d:%02d", minute, second);
	}
  
}